### PR TITLE
Try to transition to circleci2

### PR DESCRIPTION
**Pull request by kmur911**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  rubocop:
+    docker:
+      - image: ruby:2.4.4@sha256:99d10d192ac1df0873480b1e0262ff0295a15f1d97e5b7bd111a7bf61a808acc
+    steps:
+      - run: apt-get update
+      - run: apt-get install -y cmake
+      - checkout
+      - run: gem install bundler
+      - run: bundle install
+      - run: bundle exec rubocop
```
Comment (krobertson) [inline .circleci/config.yml]: Should also add one for rspec

Comment (kmur911) [reply .circleci/config.yml]: does that mean we don't need to run both separately?

File: .ruby-version [hunk 2]
```diff
@@ -1 +1 @@
-2.4.1
+2.4.4
```
Comment (itstehkman) [inline .ruby-version]: Why do we need to update .ruby-version? Will this break anything?

Comment (chwevans) [reply .ruby-version]: to add onto this, tests use ruby 2.5, should we use ruby 2.5 instead?

Comment (kmur911) [reply .ruby-version]: this was me being inconsistent, the example config i'd copied from originally used the 2.4.4 ruby image, and so i tested it with that locally. I think it's better I change it back to 2.4.1 and use that image for tests as well

**Pull request merged**